### PR TITLE
Update BE authentication routes, fix cookie saving

### PR DESCRIPTION
**Pull request by joelmackenz**
BE routes working, cookie saved upon sign-in

### CODE REVIEW

File: server/models/user.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+var mongoose = require('mongoose');
```
Comment (dproc96) [inline server/models/user.js]:
```suggestion
const mongoose = require('mongoose');
```
Can you update all the vars to const or let

File: server/routes/signup.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const initPassport = require('../passport-config')
+initPassport(passport)
+
+const User = require('../models/user')
+
+router.route('/')
+  .post((req, res, next) => {
+    if (req.body.password.length >= 6 && req.body.firstName !== null && req.body.lastName !== null) {
+    User.register(new User({firstName: req.body.firstName, lastName:req.body.lastName, email: req.body.email}), req.body.password)
```
Comment (dproc96) [inline server/routes/signup.js:13]: What is this function doing? Does the password get hashed and saved to the database?

**Review by dproc96 — COMMENTED** [review]
I'm not really clear on `passport-local-mongoose` and how secure it is or what it's actually doing. If the rest of your team is on board with using that library go ahead, but I think it's obscuring a lot of important function around verification which we want to be careful about

File: server/routes/signup.js (hunk 2 above)
Comment (joelmackenz) [reply server/routes/signup.js:13]: Yes, that's exactly right. The other fields are saved normally, the password is hashed and hash and salt values are saved to the DB.

File: server/models/user.js (hunk 1 above)
Comment (joelmackenz) [reply server/models/user.js]: My mistake. Done.

**Comment by joelmackenz** [comment]
Checked with Jess and he approved us using passport-local-mongoose for now. It is saving the salt and hash values to the DB.

File: server/routes/signin.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        {user: req.body.email},
+        process.env.JWT_SECRET,
+        {expiresIn: "1d"}
+        );
+
+      res.cookie("Token", token, {
+        httpOnly: true,
+      })
+      
+      return res.send("Login successful")
```
Comment (dproc96) [inline server/routes/signin.js]: On integrating it may be beneficial to send the user object (minus the password) in the response here and in Sign Up

**Review by dproc96 — APPROVED** [review]
Had an optional change request, but otherwise this is good to merge. Nice work!

**Pull request merged**